test/unit/pjax_fallback.js

```diff
@@ -313,6 +313,50 @@ asyncTest("POST with data object"+s, function() {
   })
 })
 
+asyncTest("GET with data array"+s, function() {
+  var frame = this.frame
+
+  this.loaded = function() {
+    equal(frame.location.pathname, "/env.html")
+    equal(frame.location.search, "?foo%5B%5D=bar&foo%5B%5D=baz")
+
+    var env = JSON.parse(frame.$("#env").text())
+    equal(env['REQUEST_METHOD'], "GET")
+    deepEqual(env['rack.request.query_hash']['foo'], ['bar', 'baz'])
+
+    start()
+  }
+
+  frame.$.pjax({
+    type: 'GET',
+    url: "env.html",
+    data: [{name: "foo[]", value: "bar"}, {name: "foo[]", value: "baz"}],
+    container: "#main"
+  })
+})
+
+asyncTest("POST with data array"+s, function() {
+  var frame = this.frame
+
+  this.loaded = function() {
+    equal(frame.location.pathname, "/env.html")
+    equal(frame.location.search, "")
+
+    var env = JSON.parse(frame.$("#env").text())
+    equal(env['REQUEST_METHOD'], "POST")
+    deepEqual(env['rack.request.form_hash']['foo'], ['bar', 'baz'])
```

Comment:
Ditto as previous comment. Comparing the entire `form_hash` instead of just one key ensures that we have absolute control over which values get sent to the server.